velox/type/TimestampConversion.cpp

```diff
@@ -812,6 +827,121 @@ Status parserError(const char* str, size_t len) {
       std::string(str, len));
 }
 
+inline bool startsWith(std::string_view str, const char* prefix) {
+  return str.rfind(prefix, 0) == 0;
+}
+
+// Spark support an id with one of the prefixes UTC+, UTC-, GMT+, GMT-, UT+ or
+// UT-, and a suffix in the formats: h[h], hh[:]mm, hh:mm:ss, hhmmss. This func
+// will validate the format of the suffix.
+// Additionally, IANA does not support seconds in the offset, so this func will
+// add the seconds offset to the input Timestamp.
+bool parseSparkTzWithPrefix(
+    std::string& timeZoneName,
+    Timestamp& ts,
+    size_t start) {
+  VELOX_DCHECK_GE(timeZoneName.length(), start + 2);
+  auto sign = timeZoneName[start];
+  auto offset = timeZoneName.substr(start + 1);
```

Comment:
Ditto. Can we use 'std::string_view'?

Reply:
The following will concatenate `offset` with other strings, and it always requires copying.

Reply:
I find the 'concat' only happens on L846. Can we specially handle it there and use string_view elsewhere? I assume the 'substr' could be avoided then.

Reply:
@rui-mo I added an `appendZeroMinutes` variable to handle this special case in the if branch.